omnigibson/systems/macro_particle_system.py

```diff
@@ -393,16 +424,33 @@ def clear(cls):
         # Clear all groups as well
         cls._particles_info = dict()
         cls._particles_local_mat = dict()
+        cls._cloth_face_ids = dict()
+
+    @classmethod
+    def remove_attachment_group(cls, group):
+        # Call super first
+        super().remove_attachment_group(group=group)
+
+        # If the group is a cloth, also remove the cloth face ids
+        if group in cls._cloth_face_ids:
+            cls._cloth_face_ids.pop(group)
+
+        return group
 
     @classmethod
     def remove_particle_by_name(cls, name):
         # Run super first
         super().remove_particle_by_name(name=name)
 
         # Remove this particle from its respective group as well
-        cls._group_particles[cls._particles_info[name]["obj"].name].pop(name)
-        cls._particles_info.pop(name)
+        group = cls._particles_info[name]["obj"].name
+        cls._group_particles[group].pop(name)
         cls._particles_local_mat.pop(name)
+        particle_info = cls._particles_info.pop(name)
+        if "face_id" in particle_info:
+            # Also remove from cloth face ids
+            face_ids = cls._cloth_face_ids[group]
+            cls._cloth_face_ids[group] = np.delete(face_ids, (face_ids == particle_info["face_id"].argmax()))
```

Comment:
Hmm is this line correct? What do you intend to do here? 

```
particle_info["face_id"].argmax()
```

Isn't `particle_info["face_id"]` just a float?

Reply:
maybe do a explicit dict look up with `_group_particles`.

Reply:
done!